tests/test_user.py

```diff
@@ -1,93 +1,82 @@
 import pytest
+import random
 from faker import Faker
 from database.models import User
 from database.database import session
+from config import Config
+from tests.fake_data import FakeUserInfo
 
 faker = Faker()
 
 
 class TestUser:
-    def test_create_user(self, test_db):
+
+    @pytest.mark.parametrize('exec_number', range(Config.NUM_TEST_EXECUTIONS))
+    def test_create_user(self, test_db, exec_number):
         """
         create user object
         get new object id from database
         check that user exists
         """
-        first_name = faker.first_name()
-        last_name = faker.last_name()
-        email = faker.email()
-        password = faker.password()
-
-        new_user_id = User.create(email=email,
-                                  first_name=first_name,
-                                  last_name=last_name,
-                                  password=password)
+        user_data = FakeUserInfo()
+        new_user_id = User.create(**user_data.to_dict())
         new_user = session.query(User).filter_by(id=new_user_id).first()
 
         assert new_user
-        assert new_user.email == email
-        assert new_user.first_name == first_name
-        assert new_user.last_name == last_name
-        assert new_user.password == password
+        assert new_user.email == user_data.email
+        assert new_user.first_name == user_data.first_name
+        assert new_user.last_name == user_data.last_name
+        assert new_user.password == user_data.password
         assert not new_user.confirmed
         assert not new_user.lists
         assert not new_user.groups
         assert not new_user.items
         assert not new_user.confirmed_on
 
-    def test_user_exists(self, test_db):
+    @pytest.mark.parametrize('exec_number', range(Config.NUM_TEST_EXECUTIONS))
+    def test_user_exists(self, test_db, exec_number):
         """
         create user object
         check that user exists
         """
 
-        first_name = faker.first_name()
-        last_name = faker.last_name()
-        email = faker.email()
-        password = faker.password()
-
-        user = User(email=email,
-                    first_name=first_name,
-                    last_name=last_name,
-                    password=password)
+        user_data = FakeUserInfo()
+        user = User(**user_data.to_dict())
 
         session.add(user)
         session.commit()
 
         assert User.exists(user.id)
 
-    def test_user_not_exist(self, test_db):
+    @pytest.mark.parametrize('exec_number', range(Config.NUM_TEST_EXECUTIONS))
+    def test_user_not_exist(self, test_db, exec_number):
         """
         check that user_id 5000 does not exist
         """
-        id = 5000
+        id = random.randint(1, 100000)
```

Comment:
If id was < 10 or so, wouldn't there be a chance there'd be a user with that id? We're not deleting out all the users before checking.

Reply:
ok I just cleared the user table before the operation